src/groupByKey.ts

```diff
@@ -2,6 +2,20 @@ interface Item {
   [key: string]: any;
 }
 
-export function groupByKey(items, field) {
-  // write code here;
+interface FilterBy {
+  [key: string]: Item[];
+}
+
+export function groupByKey(items: Item[], field: keyof Item): FilterBy {
+  const filteredByKey: FilterBy = {};
+
+  for (let i: number = 0; i < items.length; i += 1) {
+    if (items[i][field] in filteredByKey) {
```

Comment:
```suggestion
  items.forEach((item: Item) {
    if (item[field] in filteredByKey) {
```

You could use `forEach` instead of `for` loop and make your code more readable